addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -755,6 +755,253 @@ export function preserveCursor(document) {
+        newNode.attributes.setNamedItem(node.attributes[index].cloneNode());
+    }
+    node.parentNode.replaceChild(newNode, node);
+}
+const removeFormat = (node, formatInfo) => {
+    node = closestElement(node);
+    if (formatInfo.hasStyle(node)) {
+        formatInfo.removeStyle(node);
+        if (['SPAN', 'FONT'].includes(node.tagName) && !node.getAttributeNames().length) {
+            return unwrapContents(node);
+        }
+    }
+
+    if (formatInfo.isTag(node)) {
+        const attributesNames = node.getAttributeNames().filter((name)=> {
+            return name !== 'oe-zws-empty-inline';
+        });
+        if (attributesNames.length) {
+            changeTagName(node, 'span');
+        } else {
+            unwrapContents(node);
+        }
+    }
+}
+
+export const formatSelection = (editor, formatName, {applyStyle, formatProps} = {}) => {
+    const selection = editor.document.getSelection();
+    if (!selection.rangeCount) return;
+    const wasCollapsed = selection.getRangeAt(0).collapsed;
+
+    const direction = getCursorDirection(selection.anchorNode, selection.anchorOffset, selection.focusNode, selection.focusOffset);
+    getDeepRange(editor.editable, { splitText: true, select: true, correctTripleClick: true });
+
+    if (typeof applyStyle === 'undefined') {
+        applyStyle = !isSelectionFormat(editor.editable, formatName);
+    }
+
+    let zws;
+    if (wasCollapsed) {
+        if (selection.anchorNode.nodeType === Node.TEXT_NODE && selection.anchorNode.textContent === '\u200b') {
+            zws = selection.anchorNode;
+            selection.getRangeAt(0).selectNode(zws);
+        } else {
+            zws = insertAndSelectZws(selection);
+        }
+        getDeepRange(editor.editable, { splitText: true, select: true, correctTripleClick: true });
+    }
+
+    const selectedTextNodes = getSelectedNodes(editor.editable)
+        .filter(n => n.nodeType === Node.TEXT_NODE && n.nodeValue.trim().length);
+
+    const formatInfo = formatsInfos[formatName];
+    for (const selectedTextNode of selectedTextNodes) {
+        const [blockOrInlineClass, ...inlineAncestors] = ancestorsUntilBlockOrHasClass(selectedTextNode).reverse();
+
+        const firstBlockOrClassHasFormat = formatInfo.isFormatted(blockOrInlineClass, formatProps);
+        let previousAncestorHasFormat = firstBlockOrClassHasFormat;
+        let lastAncestorInlineFormat;
+
+        for (const ancestor of inlineAncestors) {
+            const ancestorIsTag = formatInfo.isTag && formatInfo.isTag(ancestor);
+            const ancestorHasStyle = formatInfo.hasStyle && formatInfo.hasStyle(ancestor, formatProps);
+            const ancestorTagIsFormated = ancestorIsTag || ancestorHasStyle;
+            const isUselessZwsSpan =
+                ancestor.tagName === 'SPAN' &&
+                ancestor.hasAttribute('oe-zws-empty-inline') &&
+                ancestor.getAttributeNames().length === 1;
```

Comment:
Otherwise, a tag could go inside that technical span and remain `forever`.
Example of a test that break without the removal of the useless span:
```
actual  : '<p>ab<span oe-zws-empty-inline=""><em oe-zws-empty-inline="">[]//zws//</em></span>cd</p>'
expected: '<p>ab<em oe-zws-empty-inline="">[]//zws//</em>cd</p>'
```

Reply:
unwrap zws and remove zws